nx-X11/lib/configure.ac

```diff
@@ -0,0 +1,283 @@
+fi
+
+AC_ARG_ENABLE(xlocale,
+              AS_HELP_STRING([--disable-xlocale],
+                [Disable Xlib locale implementation *EXPERIMENTAL*]),
+              [XLOCALE=$enableval],[XLOCALE=yes])
+
+AM_CONDITIONAL(XLOCALE, [ test x$XLOCALE = xyes ])
+if test x"$XLOCALE" = "xyes"; then
+        AC_DEFINE(XLOCALE,1,[support for X Locales])
+fi
+
+
+# This disables XLOCALEDIR.  Set it if you're using BuildLoadableXlibI18n,
+# don't have either issetugid() or getresuid(), and you need to protect
+# clients that are setgid or setuid to an id other than 0.
+AC_MSG_CHECKING([if XLOCALEDIR support should be enabled])
+AC_ARG_ENABLE(xlocaledir,
+              AS_HELP_STRING([--enable-xlocaledir],
+                [Enable XLOCALEDIR environment variable support]),
+              [ENABLE_XLOCALEDIR=$enableval],[ENABLE_XLOCALEDIR=$XLOCALEDIR_IS_SAFE])
+if test "x$ENABLE_XLOCALEDIR" = "xno"; then
+        AC_DEFINE(NO_XLOCALEDIR,1,[Disable XLOCALEDIR environment variable])
+fi
+AC_MSG_RESULT($ENABLE_XLOCALEDIR)
+
+AC_ARG_ENABLE(xf86bigfont,
+              AS_HELP_STRING([--disable-xf86bigfont],
+                [Disable XF86BigFont extension support]),
+              [XF86BIGFONT=$enableval],[XF86BIGFONT="yes"])
+if test "x$XF86BIGFONT" = "xyes"; then
+# FIXME: Skip check for proto headers for now... Requires .pc files for all the proto
+# header packages we have. Keeping that in mind for later...
+#    PKG_CHECK_MODULES(BIGFONT, [xf86bigfontproto >= 1.2.0],
+        AC_DEFINE(XF86BIGFONT,1,[Enable XF86BIGFONT extension])
+#   ,XF86BIGFONT="no")
+fi
+
+AC_ARG_ENABLE(xkb,
+              AS_HELP_STRING([--disable-xkb],
+                [Disable XKB support *EXPERIMENTAL*]),
+              [XKB=$enableval],[XKB=yes])
+
+AM_CONDITIONAL(XKB, [ test x$XKB = xyes ])
+if test x"$XKB" = "xyes"; then
+   XKBPROTO_REQUIRES="kbproto"
+   X11_REQUIRES="${X11_REQUIRES} kbproto inputproto"
+   AC_DEFINE(XKB,1,[Use XKB])
+else
+   XKBPROTO_REQUIRES=""
+fi
+AC_SUBST(XKBPROTO_REQUIRES)
+
+AC_FUNC_MMAP()
+composecache_default=$ac_cv_func_mmap_fixed_mapped
+AC_CHECK_FUNC(nl_langinfo, , [composecache_default=no])
+AC_ARG_ENABLE(composecache,
+              AS_HELP_STRING([--disable-composecache],
+                [Disable compose table cache support]),
+              [COMPOSECACHE=$enableval],[COMPOSECACHE=$composecache_default])
+if test x"$COMPOSECACHE" = "xyes"; then
+        AC_DEFINE(COMPOSECACHE,1,[Include compose table cache support])
+fi
+
+X11_DATADIR="${datadir}/nx/X11"
+AX_DEFINE_DIR(X11_DATADIR, X11_DATADIR, [Location of libNX_X11 data])
+AC_SUBST(X11_DATADIR)
+
+X11_LIBDIR="${libdir}"
+AX_DEFINE_DIR(X11_LIBDIR, X11_LIBDIR, [Location of libNX_X11 library data])
+AC_SUBST(X11_LIBDIR)
+
+# FIXME: Skip check for proto headers for now... Requires .pc files for all the proto
+# header packages we have. Keeping that in mind for later...
+#PKG_CHECK_MODULES(X11, [$X11_REQUIRES])
+X11_CFLAGS="$X11_CFLAGS $XTHREAD_CFLAGS $BASE_CFLAGS"
```

Comment:
> Hm, upstream doesn't add $BASE_CFLAGS here. Do they do that somewhere else? Or are they just not using at all for libX11?

They handle the BASE_CFLAGS via the cross build section further above in their configure.ac file. The part, I have dropped from libNX_X11 (and still want to keep it omitted).